app/frontend/src/features/profile/hooks/useUpdateHostingPreferences.test.tsx

```diff
@@ -12,29 +12,53 @@ const updateHostingPreferenceMock = service.user
   .updateHostingPreference as jest.Mock;
 
 describe("useUpdateHostingPreference hook", () => {
-  const newHostingPreferenceData = {
-    acceptsKids: false,
+  const newHostingPreferenceData: HostingPreferenceData = {
     acceptsPets: false,
     area: "",
     houseRules: "",
     lastMinute: true,
     maxGuests: null,
     smokingAllowed: 1,
     wheelchairAccessible: true,
+    hasPets: false,
+    petDetails: "",
+    hasKids: false,
+    acceptsKids: false,
+    kidDetails: "",
+    hasHousemates: false,
+    housemateDetails: "",
+    smokesAtHome: false,
+    drinkingAllowed: false,
+    drinksAtHome: false,
+    otherHostInfo: "",
+    campingOk: true,
+    parking: true,
+    parkingDetails: 3,
+    sleepingArrangement: 2,
+    sleepingDetails: "",
   };
 
   it("updates the store with the latest user hosting preference", async () => {
     addDefaultUser();
-    const newUserPref = Object.entries(newHostingPreferenceData).reduce(
-      (acc: Record<string, unknown>, [key, value]) => {
-        if (key !== "smokingAllowed") {
-          acc[key] = { value };
-        } else {
-          acc[key] = value;
+
+    const newUserPref = (Object.keys(
+      newHostingPreferenceData
+    ) as (keyof HostingPreferenceData)[]).reduce(
+      (acc, key: keyof HostingPreferenceData) => {
+        switch (key) {
+          case "smokingAllowed":
+          case "parkingDetails":
+          case "sleepingArrangement":
+            acc[key] = newHostingPreferenceData[key];
+            return acc;
+          default:
+            acc[key] = {
+              value: newHostingPreferenceData[key],
+            };
+            return acc;
         }
-        return acc;
       },
-      {}
+      {} as Record<string, unknown>
```

Comment:
What's wrong with using `Object.entries` as before? You could literally swap the switch logic you got there with the old logic and everything works fine without all the extra type assertions, which I'm not keen of doing unless absolutely needed as they are not type safe.

Reply:
I can use object.entries as well, but even then I would like to add the type assertions. Without these type assertings, the 'key' variable can contain any string and the typescript compiler won't tell you if you make a typo or if a property gets removed or renamed.